### PR TITLE
Add support for Janet

### PR DESCRIPTION
**Pull request by bakpakin**
## Description
Adds support for the [Janet](https://janet-lang.org) programming language. Fixes: janet-lang/janet#146

I don't think Janet yet meets the popularity requirements, but I thought I would open the PR since we have a working TextMate grammar and let people take a look. Looking through the search results, I didn't see any extension collisions except for some Java in [one repository](https://github.com/dejwk/janet).

## Checklist:
- [x] **I am adding a new language.**
	- [ ] **The extension of the new language is used in hundreds of repositories** _(Pending review)_
		- [Search results](https://github.com/search?q=extension%3Ajanet+NOT+nothack&type=Code)
	- [x] **I have included real-world usage samples:**
		- Source: [`janet-lang/janet`](https://github.com/janet-lang/janet/tree/5464a7a3793fe52d8ecd42b7205e00bcf3c9f634/examples)
		- License: [MIT](https://github.com/janet-lang/janet/blob/bbae43f259463c590b42c4b2d13779cbd4577cbe/LICENSE)
	- [x] **I have included a syntax highlighting grammar:**
		- Source:  [`janet-lang/vscode-janet`](https://github.com/janet-lang/vscode-janet)
		- License: [MIT](https://github.com/janet-lang/vscode-janet/blob/d6d2c34a07634521f575d9606c39411395aad70e/LICENSE)

### CODE REVIEW

**Comment by Alhadis** [comment]
> I don't think Janet yet meets the popularity requirements, but I thought I would open the PR since we have a working TextMate grammar and let people take a look.

Advertising a project is no reason for submitting a pull-request *you already know* fails to meet the documented requirements. Please resubmit once the project is approaching sufficient levels of use. Currently, there are only [336 `.janet` files](https://github.com/search?q=extension%3Ajanet+NOT+nothack&type=Code) distributed between 56 unique repositories and only 30 users (the majority of them coming from @julia-lang):

<details><summary><b>56 repositories:</b></summary>

```
ahungry/janet-code
andrewchambers/janet-redo
andrewchambers/janetsh
azaelmsousa/TransferLearning
bakpakin/littleserver
bakpakin/mendoza
bakpakin/x43bot
bimargulies/jdk-source-code
brandonchartier/html
chirino/janet
dejwk/janet
eagle518/jdk-source-code
freecores/myforthprocessor
Homarechan/HelloWorld
janet-lang/argparse
janet-lang/circlet
janet-lang/janet-lang.org
janet-lang/janet.tmLanguage
janet-lang/janet.vim
janet-lang/janet
janet-lang/janetui
janet-lang/jaylib
janet-lang/jhydro
janet-lang/json
janet-lang/juv
janet-lang/path
janet-lang/pkgs
janet-lang/sqlite3
janet-lang/webview
jfcap/janet-luajit
joy-framework/cipher
joy-framework/codec
joy-framework/joy
joy-framework/tester
joy-framework/uuid
lorelib/jdk-4_5_6
MuniyappanV/jdk-source-code
PaulBatchelor/junt
PaulBatchelor/weewiki
pauldub/janet-http-client
pepe/chidi
pepe/janet-playground
pepe/trolley
Putnam3145/janet-d
quan-nh/adventofcode
quan-nh/sync-run
RayMPerry/kitchen-sink
rduplain/hosts
rrees/janet-playground
Ruin0x11/janet-shiori
sepisoad/jurl
sepisoad/super-janet-typist
subxiang/jdk-source-code
TheRenegadeCoder/sample-programs
weimingtom/jdk-source-code
zraffo/poker_learn
```

</details>

<details><summary><b>30 users:</b></summary>

```
ahungry
andrewchambers
azaelmsousa
bakpakin
bimargulies
brandonchartier
chirino
dejwk
eagle518
freecores
Homarechan
janet-lang
jfcap
joy-framework
lorelib
MuniyappanV
PaulBatchelor
pauldub
pepe
Putnam3145
quan-nh
RayMPerry
rduplain
rrees
Ruin0x11
sepisoad
subxiang
TheRenegadeCoder
weimingtom
zraffo
```

</details>

In other words, you're a long way off.

> The grammar is at https://github.com/janet-lang/janet.tmLanguage, and has just been pulled out of the Janet repository so that is stands alone.

You have a [VSCode grammar](https://github.com/janet-lang/vscode-janet), which already includes a [`janet.tmLanguage` file](https://github.com/janet-lang/vscode-janet/blob/f51015b07e425b5e09a03908d3ba7c62dd965031/syntaxes/janet.tmLanguage). As I'm sure you must have read in our [contribution guidelines](https://github.com/github/linguist/blob/master/CONTRIBUTING.md), Linguist supports VSCode syntaxes too, meaning you can simply use the existing `vscode-janet` repository instead of creating a new one.

I'm closing this as there's no point keeping this open in the meantime.

**Comment by elimisteve** [comment]
Hi @Alhadis, Janet now has much more usage! See https://github.com/search?q=extension%3Ajanet+NOT+nothack&type=Code . Think it qualifies now?

**Comment by uvtc** [comment]
What number does that have to reach? Janet is over 1500 search results now (using that link above).

Note, I think some projects are marking their code as clojure instead of janet so that they get *some* syntax highlighting. Could that cause the numbers of .janet files to be underreported?

**Comment by JeysonFlores** [comment]
According to [linguist's contribution guide ](https://github.com/github/linguist/blob/master/CONTRIBUTING.md):

> We try only to add new extensions once they have some usage on GitHub. In most cases we prefer that each new file extension be in use in at least 200 unique :user/:repo repositories before supporting them in Linguist.

It has been almost 3 years since this PR was opened. I'm not sure if Janet has already reached the 200 repos using it but I'm pretty sure that if it hasn't, then it's pretty close. It also has reached 2,220 files all over GitHub.

**Comment by pyrmont** [comment]
@uvtc You can use [Harvester](https://github.com/Alhadis/Harvester)  to extract data from the search results page. I believe that's what @Alhadis is doing here.

**Comment by Alhadis** [comment]
@bakpakin I've fixed the merge conflicts with your branch and updated a bunch of fields. See https://github.com/bakpakin/linguist/pull/1.

Your branch wasn't writeable by Linguist maintainers, so I had to send you a pull-request instead.

**Comment by elimisteve** [comment]
> In most cases we prefer that each new file extension be in use in at least 200 unique :user/:repo repositories before supporting them in Linguist.

> Still a bit light on usage, ... . Out of 2,343 results, I was able to compile a list of 2,269 files, distributed between 371 repos by 121 users.

371 >= 200, last I checked, so doesn't Janet meet the threshold?

**Comment by pyrmont** [comment]
@lildude Thanks for cleaning up the PR. Is this now in a mergable state? Or does it require further work?

**Review by lildude — APPROVED** [review]
Approving this based on the temporary requirements detailed in https://github.com/github/linguist/issues/5756

**Comment by elimisteve** [comment]
@lildude Thank you! 🥇
